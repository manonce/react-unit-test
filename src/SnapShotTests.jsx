import React, { Component } from "react";

class SnapShotTests extends Component {
  state = {
    showName: false
  };
  toggleShowName = () => {
    const { showName } = this.state;
    this.setState({
      showName: !showName
    });
  };
  render() {
    const { showName } = this.state;
    return (
      <div className="App">
        {Date.now()}
        
      </div>
    );
  }
}

export default SnapShotTests;
