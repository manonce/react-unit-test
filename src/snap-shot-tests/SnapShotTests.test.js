import React from "react";
import renderer from "react-test-renderer";
import { shallow } from 'enzyme';
import SnapShotTests from "../SnapShotTests";

it("renders correctly", () => {
  const app = shallow(<SnapShotTests/>);
  // Date.now = jest.fn(()=>1234)
  const tree = renderer.create(<SnapShotTests />).toJSON();
  console.log(tree);
  expect(tree).toMatchSnapshot({
  children: [ expect.any(String) ] });
  // expect(tree).toMatchSnapshot(<div
  //   className="App"
  // >
  //   {expect.any(String)}
  // </div>)
  // app.find('button').simulate('click');
  // expect(tree).toMatchSnapshot();
});
