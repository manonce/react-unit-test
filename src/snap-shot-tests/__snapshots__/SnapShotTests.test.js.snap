// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "children": Array [
    Any<String>,
  ],
  "props": Object {
    "className": "App",
  },
  "type": "div",
}
`;
